### ISSUE TITLE
 There may be data leakage when you train the cross-subject model with your code.

### ISSUE THREAD

**Issue by bobwangPKU** [dongyangli-del/EEG_Image_decode#7]
When you trained your model for the leave-one test subject, you did not randomly initialize the model, but instead used the model trained for the previous test subject. [here.](https://github.com/dongyangli-del/EEG_Image_decode/blob/8f4c5e57f3a57e18792ad088cdf2f2eb7cf34e37/Retrieval/ATM_S_insubject_retrieval.py#L539) This means that your model has seen this test subject before. This is also the reason why the cross subject model achieved the highest performance in the first epoch. In fact, if you test the 0th epoch, the results will be even better.

**Comment by dongyangli-del** [comment]
@bobwangPKU Hello, thank you for your attention to our work and for pointing out the hidden danger that cross-subject training data exposure bias may lead to falsely high results.

We confirmed this issue after a preliminary code review. We are correcting the experimental results, and we will update the results to the github remote repository as soon as possible. 

We will keep in touch with you and thank you again!

**Comment by dongyangli-del** [comment]
@bobwangPKU 
Thank you for pointing out our problem. We immediately modified the code to ensure random initialization of the model before training and testing on different subjects.

Preliminary results indicate:
1. We found that the problem of high accuracy at the 0th epoch in cross-subject training and testing is indeed caused by invisible exposure bias. The same problem also exists in within-subject training and testing, but we are not sure yet. Does exposure bias bring gains or negative effects to the subjects?

2. The accuracy of the corrected code on the cross-subject task is not much different from the results of the leave-one-subject test reported in the table in the appendix of the paper ([https://arxiv.org/pdf/2403.07721.pdf](https://arxiv.org/pdf/2403.07721.pdf)). We speculate that this is because the table takes the average test accuracy of the last 10 test accuracy epochs, and the last 10 epoch may be less affected by this subject exposure bias.

3. When comparing other methods, we also mistakenly placed the random initialization of the model outside the loop, so we are urgently reproducing other methods with the correct code and updating our GitHub repository and the results in the arXiv article.

![image](https://github.com/dongyangli-del/EEG_Image_decode/assets/62006079/ae6ebe08-a8ac-4ff2-9bec-6d0db8e00eec)

**Comment by dongyangli-del** [comment]
@bobwangPKU  Hi, do you have any suggestions for our feedback? More constructive suggestions for improvements are welcome.

**Comment by bobwangPKU** [comment]
Thank you for your immediate response to my confusion. A really nice work！I need to further read your code and article to learn your ideas. 
For your question, the exposure bias will have a positive impact on the results, but the bias of the model weights on the test subjects may gradually decrease as the training epoch increases. Therefore, if the last 10 epochs are selected as the results, this impact may not be significant.

**Comment by dongyangli-del** [comment]
Thanks for your reply, I will update further with more experimental results.
